### PR TITLE
replaced sending logs from `Sentry` to `Hastebin`

### PR DESCRIPTION
**Pull request by Hidanio**

### CODE REVIEW

File: src/main/kotlin/com/vk/admstorm/services/HastebinService.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.vk.admstorm.services
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.vk.admstorm.CommandRunner
+import com.vk.admstorm.env.Env
+
+@Service
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/services/HastebinService.kt:8]:
```suggestion
@Service(Service.Level.PROJECT)
```

File: src/main/kotlin/com/vk/admstorm/services/HastebinService.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+class HastebinService() {
+    companion object {
+        fun createHaste(project: Project, data: String): String {
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/services/HastebinService.kt:11]: take it out of the `companion object`

File: src/main/kotlin/com/vk/admstorm/services/SentryService.kt
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/services/SentryService.kt:1]: Forgot to delete `sendIdeaLog`

File: src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt [hunk 3]
```diff
@@ -3,7 +3,7 @@ package com.vk.admstorm.ui
 import com.intellij.ui.IconManager
 
 object AdmIcons {
-    private fun icon(name: String) = IconManager.getInstance().getIcon(name, javaClass)
+    private fun icon(name: String) = IconManager.getInstance().getIcon(name, javaClass.classLoader)
```
Comment (Danil42Russia) [inline src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt:6]: why?

**Comment by Danil42Russia** [comment]
@Hidanio, i updated the log, pull up the change for yourself

File: src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt (hunk 3 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt:6]: .getIcon(name, javaClass) deprecated
I updated it

File: src/main/kotlin/com/vk/admstorm/actions/SendLogsToHastebinAction.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package com.vk.admstorm.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.vk.admstorm.notifications.AdmNotification
+import com.vk.admstorm.services.HastebinService
+import com.vk.admstorm.ui.AdmIcons
+import com.vk.admstorm.utils.MyUtils
+import com.vk.admstorm.utils.MyUtils.readIdeaLogFile
+class SendLogsToHastebinAction : AdmActionBase() {
```
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/actions/SendLogsToHastebinAction.kt:9]: Done

File: src/main/kotlin/com/vk/admstorm/utils/MyUtils.kt [hunk 5]
```diff
@@ -94,9 +98,42 @@ object MyUtils {
         return result.toString().removeSuffix("\n")
     }
 
-    fun createHaste(project: Project, data: String): String {
-        val output = data.replace("\"", "\\\"").replace("$", "\\$")
-        return CommandRunner.runRemotely(project, "echo \"$output\" | ${Env.data.pasteBinCommand}").stdout
+    fun readIdeaLogFile(full: Boolean = false): String {
```
[the remark this answers is not in the document]
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/utils/MyUtils.kt:101]: Perhaps in the future we may need this, I replaced the body of the function "readIdeaLogFileByte" with the call "return readIdeaLogFile(full).toByteArray(StandardCharsets.UTF_8)"

File: src/main/kotlin/com/vk/admstorm/services/HastebinService.kt (hunk 2 above)
Comment (Hidanio) [reply src/main/kotlin/com/vk/admstorm/services/HastebinService.kt]: Done

File: src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt (hunk 3 above)
Comment (Danil42Russia) [reply src/main/kotlin/com/vk/admstorm/ui/AdmIcons.kt:6]: ok

**Review by Danil42Russia — APPROVED** [review]

**Pull request merged**